app/lib/profile/models/hive/diplotype.dart

```diff
@@ -39,10 +42,19 @@ class Diplotype {
   }
 }
 
+
 extension FilteredList on List<Diplotype> {
   List<Diplotype> filterValidDiplotypes() {
     final acceptedResultTypes = ['Diplotype'];
     return where((element) => acceptedResultTypes.contains(element.resultType))
         .toList();
   }
 }
+
+class Diplotypes {
+  // assumes http reponse from lab server
+  static List<Diplotype> fromHTTPResponse(Response resp) {
+    final json = jsonDecode(resp.body)['diplotypes'] as List<dynamic>;
+    return json.map<Diplotype>(Diplotype.fromJson).toList();
+  }
+}
```

Comment:
I would integrate this functionality in the Diplotype class I think. Don't see the point of making a different class.

Reply:
NNP